hpe3parclient/client.py

```diff
@@ -4208,3 +4208,74 @@ def resyncPhysicalCopy(self, volume_name):
         info = {'action': self.RESYNC_PHYSICAL_COPY}
         response = self.http.put("/volumes/%s" % (volume_name), body=info)
         return response[1]
+    
+    def createSchedule(self, schedule_name, task, taskfreq='@hourly'):
+        """Create Schedule for volume snapshot.
+        :param schedule_name - The name of the schedule
+        :type - string
+        :param task - command to for which schedule is created
+        :type - string
+        :param taskfreq - frequency of schedule 
+        :type - string
+        """
+        cmd = ['createsched']
+        cmd.append("\""+task+"\"")
+        if '@' not in taskfreq:
+           cmd.append("\""+taskfreq+"\"")
+        else:
+           cmd.append(taskfreq)
+        cmd.append(schedule_name)
+        try:
+            resp = self._run(cmd)
+
+            err_resp = self.check_response(resp)
+            if err_resp:
+                err = (("Create snapschedule failed Error is"
+                         " '%(err_resp)s' ") %
+                       {'err_resp': err_resp})
+                raise exceptions.SSHException(reason=err)
+        except exceptions.SSHException as ex:
+            raise exceptions.SSHException(reason=ex)        
+        
+
+    def deleteSchedule(self, schedule_name):
+        """Delete Schedule 
+        :param schedule_name - The name of the schedule to delete
+        :type - string
+        """
+        cmd = ['removesched', '-f', schedule_name]
+        try:
+            resp = self._run(cmd)
+
+            err_resp = self.check_response(resp)            
+            if err_resp:
+                err = (("Delete snapschedule failed Error is"
+                         " '%(err_resp)s' ") %
+                       {'err_resp': err_resp})
+                raise exceptions.SSHException(reason=err)
+        except exceptions.SSHException as ex:
+            raise exceptions.SSHException(reason=ex)
+      
+    def check_response(self, resp):
+        for r in resp:
+            if 'Error' in str.lower(r) or 'Invalid' in str.lower(r):
+                err_resp = r.strip()
+                return err_resp
+        return
+
+    def getSchedule(self, schedule_name):
+        """Get Schedule
+        :param schedule_name - The name of the schedule to get information 
+        :type - string
+        """
+        cmd = ['showsched ', schedule_name]
+        try:
+           result = self._run(cmd)
+           if 'No scheduled tasks ' in result :
+              err =  (("Getting schedule failed Error is"
+                         " '%(result)s' ") %
+                       {'err_resp': result})
+              raise exceptions.SSHException(reason=err)
```

Comment:
Why are we raising an exception here? We could just return nothing saying no such task.